src/plugins/kibana_usage_collection/server/collectors/core/core_usage_collector.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { UsageCollectionSetup } from 'src/plugins/usage_collection/server';
+import {
+  CoreUsageData,
+  CoreUsageDataStart,
+  // eslint-disable-next-line @kbn/eslint/no-restricted-paths
+} from '../../../../../core/server/core_usage_data/types';
+// telemetry_check is unable to check types which are re-exported, it only
+// works when pointed to the typescript file where the type was defined.
+import { KIBANA_STATS_TYPE } from '../../../common/constants';
+
+export function getCoreUsageCollector(
+  usageCollection: UsageCollectionSetup,
+  getCoreUsageDataService: () => CoreUsageDataStart
+) {
+  return usageCollection.makeUsageCollector<CoreUsageData, { core: CoreUsageData }>({
+    type: 'core',
+    isReady: () => true,
```

Comment:
I would change this to validate the getter `getCoreUsageDataService` now returns anything but undefined.
```suggestion
    isReady: () => typeof getCoreUsageDataService() !== 'undefined',
```
There could be a race condition, but since it will be defined after the `start` method (and it should not change), the chances of the race condition actually affecting us are very limited.